### ISSUE TITLE
Use GitService in PushToRemoteStage

### ISSUE THREAD

**Issue by michaelsauter** [opendevstack/ods-jenkins-shared-library#256]
https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/quickstarter/PushToRemoteStage.groovy#L18-L44 contains a lot of Git related things that can be split into parts and moved to the GitService.

**Comment by clemensutschig** [comment]
agreed, especially given all the work you have done MRO for git related stuff :)

**Comment by clemensutschig** [comment]
god, this is unbelievably painfull

**Comment by michaelsauter** [comment]
Didn't get to it yet. Relatively low prio as the current implementation works as-is, therefore taking it off from v3.
